pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -783,6 +787,49 @@ func (r *ReconcileClusterDeployment) updateClusterDeploymentStatus(cd *hivev1.Cl
 	return nil
 }
 
+func (r *ReconcileClusterDeployment) generateAdminKubeconfigSecret(cd *hivev1.ClusterDeployment, rawSecret *corev1.Secret, cdLog log.FieldLogger) (string, error) {
+	var hashPart string
+	if len(controllerutils.AdditionalCAHash()) == 0 {
+		hashPart = "original"
+	} else {
+		additionalCAHash := controllerutils.AdditionalCAHash()
+		hashPart = additionalCAHash[:20]
+	}
+	name := fmt.Sprintf("%s-%s", rawSecret.Name, hashPart)
```

Comment:
so every additional CA change results in a new secret, but we don't cleanup the old ones. That is probably ok but is there a reason we can't just use a constant secret name and avoid leaking them?

Reply:
The reason I use the name with the hash is that I don't have to worry about updating an out of date secret. If the secret exists with the right hash, then I just skip creating it.